pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java

```diff
@@ -383,7 +383,12 @@
   public static class Segment {
     public static class Realtime {
       public enum Status {
-        IN_PROGRESS, DONE
+        // Means the segment is not completed and still consuming stream data
+        IN_PROGRESS,
+        // Means the segment is completed and sealed by some Pinot server and uploaded
+        DONE,
+        // Means the segment is uploaded to a Pinot controller by an external party
+        UPLOADED
```

Comment:
Done.

Reply:
You should only add enums at the end. But anyways, this is a deeper change. I was just asking of the other reviewers. Let us revert it and stick to your PR for now.
@npawar  and @Jackie-Jiang  may have an opinion. We can discuss separately.

Reply:
I feel `CONSUMING` and `COMPLETED` is more clear, and consistent with the term we used in the segment assignment strategy

Reply:
Removed the two enums.

Reply:
@Jackie-Jiang agreed, just that it will need a migration of the existing metadata and/or code to scan both. Best to keep it out of this PR. @chenboat  will be great if you can file an issue for this.

Reply:
https://github.com/apache/incubator-pinot/issues/6777 filed for this issue.